src/main.rs

```diff
@@ -353,71 +38,49 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 			scenario,
 			send_threshold,
 			remark,
-		} => {
-			let recipe = remark.map_or_else(
-				|| TransactionRecipe::transfer(),
-				|size| TransactionRecipe::remark(size),
-			);
-			match chain {
-				ChainType::Fake => {
-					let sink = FakeTransactionSink::new();
-					let builder = FakeTransactionBuilder::new();
-					execute_scenario(
-						sink,
-						builder,
-						scenario,
-						*unwatched,
-						*send_threshold as usize,
-						&recipe,
-					)
-					.await;
-				},
-				ChainType::Eth => {
-					let transaction_monitor =
-						if *block_monitor { Some(BlockMonitor::new(ws).await) } else { None };
+		} => match chain {
+			ChainType::Fake => {
+				unimplemented!()
+			},
+			ChainType::Eth => {
+				let mut scenario_builder = ScenarioBuilder::default()
+					.with_rpc_uri(ws.to_string())
+					.with_scenario_type(scenario.clone())
+					.with_block_monitoring()
+					.with_chain_type(chain.clone())
+					.with_send_threshold(*send_threshold as usize);
+
+				if !unwatched {
+					scenario_builder = scenario_builder.with_watched_txs();
+				}
 
-					let def = scenario.get_accounts_description();
-					let sink = EthTransactionsSink::new_with_uri_with_accounts_description(
-						ws,
-						def,
-						generate_ecdsa_keypair,
-						transaction_monitor,
-					)
-					.await;
-					let builder = EthTransactionBuilder::new();
-					execute_scenario(
-						sink,
-						builder,
-						scenario,
-						*unwatched,
-						*send_threshold as usize,
-						&recipe,
-					)
-					.await;
-				},
-				ChainType::Sub => {
-					let transaction_monitor =
-						if *block_monitor { Some(BlockMonitor::new(ws).await) } else { None };
-					let def = scenario.get_accounts_description();
-					let sink = SubstrateTransactionsSink::new_with_uri_with_accounts_description(
-						ws,
-						def,
-						generate_sr25519_keypair,
-						transaction_monitor,
-					)
-					.await;
-					let builder = SubstrateTransactionBuilder::new();
-					execute_scenario(
-						sink,
-						builder,
-						scenario,
-						*unwatched,
-						*send_threshold as usize,
-						&recipe,
-					)
-					.await;
-				},
-			};
+				if let Some(inner) = remark {
+					scenario_builder = scenario_builder.with_remark_tx_recipe(*inner);
+				}
+
+				let scenario_executor = scenario_builder.build().await;
+				let _ =
+					scenario_executor.execute_txs::<DefaultTxTask<SubstrateTransaction>>().await;
+			},
+			ChainType::Sub => {
+				let mut scenario_builder = ScenarioBuilder::default()
+					.with_rpc_uri(ws.to_string())
+					.with_scenario_type(scenario.clone())
+					.with_block_monitoring()
```

Comment:
This should take bool: 
```
.with_block_monitoring(block_monitoring)
```

Reply:
Hmm, I actually imagined it as we either call the setter (setting it to `true`), or we don't (initialized with false and kept that way). I was thinking we're not interested in changing it once we set it to `true`, so no need to give the option to "unset it". WDYT?

Reply:
Addressed your initial request here: [9f14440](https://github.com/michalkucharczyk/tx-test-tool/pull/22/commits/9f14440a0747fa8f44b382607d6bcc36649465e7).